### DISCUSSION TITLE
How do I format a drive?

### DISCUSSION THREAD

**Discussion by KernelFox0** [CosmosOS/Cosmos#2135] [Q&A]
Sys.FileSystem.VFS.VFSManager.Format("0", "FAT32", true) worked on earlier versions of cosmos but it's not working in the latest dev kit.

**Comment by ghost** [comment]
^

**Reply by tails1154** [reply]
that is not working

**Reply by randomusert** [reply]
Check docs

**Comment by Szymekk44** [comment]
Hi! first you can check if there are any existing partitions, and delete them using Clear()
```
Kernel.fs.Disks[0].Clear();
Kernel.fs.Disks[0].CreatePartition((int)(Kernel.fs.Disks[0].Size / (1024 * 1024))); //Its in MB and Kernel.fs.Disks[0].Size is in bytes
Kernel.fs.Disks[0].FormatPartition(0, "FAT32", true); 
```
for more please check https://cosmosos.github.io/articles/Kernel/VFS.html
